pkg/server/admin.go

```diff
@@ -1291,21 +1290,30 @@ func (s *adminServer) DecommissionStatus(
 
 	// Compute the replica counts for the target nodes only. This map doubles as
 	// a lookup table to check whether we care about a given node.
-	replicaCounts := make(map[roachpb.NodeID]int64)
-	for _, nodeID := range nodeIDs {
-		replicaCounts[nodeID] = math.MaxInt64
-	}
-
-	for _, nodeStatus := range ns.Nodes {
-		nodeID := nodeStatus.Desc.NodeID
-		if _, ok := replicaCounts[nodeID]; !ok {
-			continue // not interested in this node
-		}
-		var replicas float64
-		for _, storeStatus := range nodeStatus.StoreStatuses {
-			replicas += storeStatus.Metrics["replicas"]
-		}
-		replicaCounts[nodeID] = int64(replicas)
+	var replicaCounts map[roachpb.NodeID]int64
+	if err := s.server.db.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
+		const pageSize = 10000
+		replicaCounts = make(map[roachpb.NodeID]int64)
+		for _, nodeID := range nodeIDs {
+			replicaCounts[nodeID] = 0
+		}
+		return txn.Iterate(ctx, keys.Meta2Prefix, keys.MetaMax, pageSize,
```

Comment:
Scanning meta2 is not sufficient - you need to scan meta1 as well to ensure that all meta2 replicas have been moved.